crates/oxc_linter/src/rules/jsx_a11y/no_aria_hidden_on_focusable.rs

```diff
@@ -0,0 +1,180 @@
+            None => true,
+            _ => false,
+        },
+        _ => false,
+    }
+}
+
+/// Determines if a JSX element is focusable.
+///
+/// According to the [W3C's DOM Level 2 HTML specification](https://www.w3.org/TR/DOM-Level-2-HTML/html.html), the elements that are focusable are:
+/// - `<a>`, `<area>` with an `href` attribute
+/// - `<button>`, `<input>`, `<select>`, `<textarea>` unless they are disabled
+/// - Any element with a `tabIndex` of zero or greater
+///
+/// This function checks the passed `JSXOpeningElement` against these criteria to determine
+/// if it is focusable.
+///
+/// # Arguments
+///
+/// * `element` - A reference to the JSXOpeningElement to check
+///
+/// # Returns
+///
+/// `true` if the element is focusable, `false` otherwise.
+fn is_focusable(element: &JSXOpeningElement) -> bool {
+    let tag_name = match &element.name {
+        JSXElementName::Identifier(JSXIdentifier { name, .. }) => name.as_str(),
+        _ => return false,
+    };
+
+    let has_tab_index = element.attributes.iter().any(|attr| match attr {
+        JSXAttributeItem::Attribute(attr) => match &attr.name {
+            JSXAttributeName::Identifier(JSXIdentifier { name, .. }) => {
+                name.as_str() == "tabIndex" || name.as_str() == "tabindex"
+            }
+            _ => false,
+        },
+        _ => false,
+    });
+
+    if has_tab_index {
+        return element.attributes.iter().any(|attr| match attr {
+            JSXAttributeItem::Attribute(attr) => match &attr.name {
+                JSXAttributeName::Identifier(JSXIdentifier { name, .. })
+                    if (name.as_str() == "tabIndex" || name.as_str() == "tabindex") =>
+                {
+                    match &attr.value {
+                        Some(JSXAttributeValue::StringLiteral(s)) => {
+                            s.value.parse::<i32>().ok().filter(|&num| num >= 0).is_some()
+                        }
+                        Some(JSXAttributeValue::ExpressionContainer(JSXExpressionContainer {
+                            expression: JSXExpression::Expression(expr),
+                            ..
+                        })) => match expr {
+                            Expression::NumberLiteral(num) => num.value as i32 >= 0,
+                            _ => false,
+                        },
+                        _ => false,
+                    }
+                }
+                _ => false,
+            },
+            _ => false,
+        });
+    }
+
+    match tag_name {
+        "a" | "area" => element.attributes.iter().any(|attr| {
+            if let JSXAttributeItem::Attribute(attr) = attr {
+                match &attr.name {
+                    JSXAttributeName::Identifier(JSXIdentifier { name, .. }) => {
+                        name.as_str() == "href"
+                    }
+                    _ => false,
+                }
+            } else {
+                false
+            }
+        }),
```

Comment:
use `has_jsx_prop_lowercase` instead?

Reply:
Thanks! fixed in https://github.com/oxc-project/oxc/pull/1795/commits/1da727200c31cb1f45751aeccd5ff5eb95176700